xbmc/addons/AddonInfo.h

```diff
@@ -81,16 +84,128 @@ namespace ADDON
     ADDON_MAX
   } TYPE;
 
+  struct extValue
+  {
+    extValue(const std::string& strValue) : str(strValue) { }
+    const std::string asString()  const { return str; }
+    const bool asBoolean() const { return StringUtils::EqualsNoCase(str, "true"); }
+    const int asInteger() const { return atoi(str.c_str()); }
+    const float asFloat() const { return static_cast<float>(atof(str.c_str())); }
+    const bool empty() const { return str.empty(); }
+    const std::string str;
+  };
+
+  class CAddonExtensions;
+  typedef std::vector<std::pair<std::string, CAddonExtensions> > EXT_ELEMENTS;
+  typedef std::vector<std::pair<std::string, extValue> > EXT_VALUE;
+  typedef std::vector<std::pair<std::string, EXT_VALUE> > EXT_VALUES;
+
+  class CAddonBuilder;
+
+  class CAddonExtensions
+  {
+  public:
+    CAddonExtensions() { }
+    ~CAddonExtensions() { }
+    bool ParseExtension(const TiXmlElement* element);
+
+    const extValue GetValue(const std::string& id) const;
+    const EXT_VALUES& GetValues() const;
+    const CAddonExtensions* GetElement(const std::string& id) const;
+    const EXT_ELEMENTS GetElements(const std::string& id = "") const;
+
+    void Insert(const std::string& id, const std::string& value);
+
+  private:
+    std::string m_point;
+    EXT_VALUES m_values;
+    EXT_ELEMENTS m_childs;
+  };
+
+  class CAddonInfo;
```

Comment:
Is for the contructor on CAddonType and also is CAddonType used from CAddonInfo, thats why the decl was needed.